readthedocs/oauth/utils.py

```diff
@@ -0,0 +1,42 @@
+import logging
+
+from django.contrib import messages
+from django.utils.translation import ugettext_lazy as _
+
+from readthedocs.oauth.services import registry
+
+log = logging.getLogger(__name__)
+
+
+def attach_webhook(project, request=None):
+    """Add post-commit hook on project import"""
+
+    for service_cls in registry:
+        if service_cls.is_project_service(project):
+            service = service_cls
+            break
+    else:
+        return None
+
+    user_accounts = service.for_user(request.user)
+    for account in user_accounts:
+        success, resp = account.setup_webhook(project)
+        if success:
+            messages.success(request, _('Webhook activated'))
+            project.has_valid_webhook = True
+            project.save()
+            break
```

Comment:
`if user_accounts` would only catch the case where accounts are returned from `for_user`, it seems we should be testing for aggregate return of each `account.setup_webhook` as well.